src/config.rs

```diff
@@ -13,39 +13,48 @@
 
 use crate::{util::ReadableSize, Result};
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum RecoveryMode {
+    TolerateCorruptedTailRecords = 0,
+    AbsoluteConsistency = 1,
+}
+
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct Config {
     pub dir: String,
-    pub recovery_mode: i32,
+    pub recovery_mode: RecoveryMode,
     pub bytes_per_sync: ReadableSize,
     pub target_file_size: ReadableSize,
-    pub cache_size_limit: ReadableSize,
-    pub total_size_limit: ReadableSize,
 
-    // Use raftstore.cfg.raft_log_gc_threshold
-    #[doc(hidden)]
-    #[serde(skip_serializing)]
-    pub compact_threshold: usize,
+    /// Only purge if disk file size is greater than `purge_threshold`.
+    pub purge_threshold: ReadableSize,
+
+    /// Only purge if garbage is greater than `garbage_threshold`.
+    pub garbage_threshold: f64,
+
+    /// Total size limit to cache log entries.
+    ///
+    /// FIXME: it doesn't make effect currently.
+    pub cache_limit: ReadableSize,
```

Comment:
Seems this parameter is not used?

Reply:
Yes, I will make it effect later, in an another PR.